java/registry/src/main/java/dev/sunbirdrc/registry/dao/VertexReader.java

```diff
@@ -346,6 +348,12 @@ private ArrayNode expandChildObject(ObjectNode entityNode, List<String> processe
                                     if (ovalue != null) {
                                         resultArr.add(ovalue);
                                     } else {
+                                        if (node.isObject()) {
+                                            Map<String, Object> deletedNodeRef = JSONUtil.convertJsonNodeToMap(node);
```

Comment:
what are these 4 lines for? these looks similar to `JsonNode refactorNode = JsonNodeFactory.instance.objectNode().put(uuidPropertyName, uuidPropertyValue);`

Reply:
This is the current logic : 

step 1:  we map all the uuids which are not deleted - uuidReferenceMap (uuid ---> referenceNode) 
       Note: ( this excludes the nodes which are deleted or status as false )
Step 2: we recursively traverse from the root node and keep expanding the children based on the depth. 

While expanding the children we always have the reference node as an object in the form --->  {osid : value }. Weather you are trying to expand an array or object. The reference is always an objectNode (cause we are using an arrayNode table where we store the internal array mappings). 

And when all the child nodes are deleted the logic returns the reference node itself while fetching. To avoid this we have to either return [null, null ] or [ {osid : deltedNode}, {osid, deletedNode}] to resolve this issue. The above four lines return the deleted referenceNode in the array.